### PR TITLE
Improve mobile ux & css

### PR DESCRIPTION
**Pull request by zhangtianli2006**

### CODE REVIEW

**Comment by zhangtianli2006** [comment]
Rendering problem on Chrome, CSS `display: -webkit-box` problem, I'll fix it tomorrow.

File: src/assets/css/basic.css [hunk 1]
```diff
@@ -149,7 +164,7 @@ h3::before {
 }
 
 .highzindex {
-    z-index: 10000 !important;
+    z-index: 1000000000000 !important;
```
Comment (szdytom) [inline src/assets/css/basic.css:167]: Don't do this, you'd better start a wiki page to talk about z-index or else the value will increase ulimited until overflow

File: src/components/lib/jumpTo.vue [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      jumpToProblem: null
+    };
+  },
+  methods: {
+    jump() {
+      this.$router.push('/problem/' + this.jumpToProblem);
+    }
+  }
+};
+</script>
```
Comment (szdytom) [inline src/components/lib/jumpTo.vue:31]: Make names **short** but **clear**, try sth like `JumpToProblem.vue`

File: src/components/lib/jumpTo.vue [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    >
+      <template slot="prepend">#.</template>
+      <el-button slot="append" icon="el-icon-top-right" @click="jump"></el-button>
+    </el-input>
+  </el-card>
+</template>
+
+<script>
+export default {
+  name: 'jumpTo',
```
Comment (szdytom) [inline src/components/lib/jumpTo.vue:19]: we use large-camel-case like `LargeCamelCase` or `JumpToProblem` for all components

File: src/components/page/navbar.vue [hunk 4]
```diff
@@ -69,7 +69,7 @@ export default {
 }
 
 #nav {
-    z-index: 1000;
+    z-index: 1000000;
```
Comment (szdytom) [inline src/components/page/navbar.vue:72]: I think we can add a `z-index.css` like this:
```css
.low-z-index {
    z-index: 100;
}
.high-z-index {
    z-index: 1000;
}
.top-z-index {
    z-index: 100000;
}
```

File: src/components/problem/content.vue [hunk 5]
```diff
@@ -18,7 +20,7 @@
       </div>
       <div id="pannel" v-if="!isWider">
         <div id="tools">
-          <el-card>
+          <el-card v-if="showSmallToolBar || !smallScreen">
```
Comment (szdytom) [inline src/components/problem/content.vue:23]: You are getting this file more and more complex... That's not good.
Well... I'll teach you how to refactory this later we back to school.

File: src/components/problem/edit.vue [hunk 6]
```diff
@@ -1,44 +1,84 @@
...
+        <el-col :span="4" class="center-text">
+          MB
+        </el-col>
+      </el-row>
+    </el-card>
+    <el-card v-loading="contentLoading" class="item">
+      <div slot="header" class="clearfix"><i class="el-icon-document" /> Content</div>
+      <MarkdownEditor v-model="mdContent" />
+    </el-card>
+  </div>
```
Comment (szdytom) [inline src/components/problem/edit.vue:81]: This file is too complex too, **please add some comment to it**.
Also, you can refactor this file by split different logic into different files, For example:

- `edit_mobile.vue`
- `edit_desktop.vue`

or even:

- `EditTags.vue`
- `EditContent.vue`
- `EditNameAndPID.vue`

File: src/components/page/navbar.vue (hunk 4 above)
Comment (zhangtianli2006) [reply src/components/page/navbar.vue:72]: Bright idea

File: src/components/problem/content.vue (hunk 5 above)
Comment (zhangtianli2006) [reply src/components/problem/content.vue:23]: The real COMPLEX comes after I tell you the huge and UNFIXABLE bug of list tag, wait for it

File: src/components/app/editor.vue [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+      Code Editor
+    </div>
+    <codeMirror />
+  </el-card>
+</template>
+
+<script>
+import codeMirror from './../lib/editor.vue';
+
+export default {
```
Comment (szdytom) [inline src/components/app/editor.vue:14]:
```javascript
export default {
  name: 'CodeEditorContainer',
  ...
}
```

File: src/components/lib/editor.vue [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+        continueLineComment: false
+      });
+    }
+  },
+  mounted() {
+    this.loadEditor();
+  },
+  destroyed() {
+    this.$router.go(0);
+  }
```
Comment (szdytom) [inline src/components/lib/editor.vue:62]: What is it?

File: src/main.js [hunk 9]
```diff
@@ -43,4 +43,4 @@ new Vue({
   router,
   store,
   render: h => h(App),
-}).$mount('#app');
\ No newline at end of file
+}).$mount('#app');
```
Comment (szdytom) [inline src/main.js:46]: What is this?

File: src/components/lib/confirm.vue [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+        <el-button type="primary" size="mini" @click="confirmed" :disabled="!confirmAnswerCorrect">Confirm</el-button>
+      </div>
+      <el-button :type="this.buttonType" slot="reference" style="margin-left: 10px;">{{buttonName}}</el-button>
+    </el-popover>
+  </span>
+</template>
+
+<script>
+export default {
+  name: 'Confirm',
```
Comment (szdytom) [inline src/components/lib/confirm.vue:28]:
```suggestion
  name: 'ConfirmDelete',
```

File: src/components/lib/tag.vue [hunk 11]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <span :v-if="showTag" class="seg-tag" :style="style">
+  <span class="seg-tag" :style="style">
```
Comment (szdytom) [inline src/components/lib/tag.vue:2]:
```suggestion
  <span class="segment-tag" :style="style">
```

File: src/components/lib/tag.vue [hunk 12]
```diff
@@ -31,35 +31,35 @@ export default {
...
     }
-    this.showTag = true;
+    this.style += 'height: ' + this.height + 'px;';
+    this.style += 'line-height: ' + String(this.height - 2) + 'px;';
   }
 };
 </script>
 
 <style scoped>
 .seg-tag {
```
Comment (szdytom) [inline src/components/lib/tag.vue:58]:
```suggestion
.segment-tag {
```

File: src/components/user/register.vue [hunk 13]
```diff
@@ -1,27 +1,59 @@
...
         <captcha ref="captchaElement" class="margin-bottom" />
         <el-form-item>
-          <el-button type="primary" v-on:click="onSubmit();" :loading="buttonLoading">Register</el-button>
+          <el-button
+            type="primary"
+            v-on:click="onSubmit();"
+            :loading="buttonLoading"
+          >
+            Register
+          </el-button>
```
Comment (szdytom) [inline src/components/user/register.vue:56]:
```suggestion
          <el-button
            type="primary"
            v-on:click="onSubmit();"
            :loading="buttonLoading"
          >Register</el-button>
```

Maybe shortter

File: src/main.js [hunk 14]
```diff
@@ -39,8 +39,8 @@ import './StringPrototype';
 
 Vue.config.productionTip = false;
 
-new Vue({
+window.that = new Vue({
```
Comment (szdytom) [inline src/main.js:42]:
```suggestion
new Vue({
```

forget to delete the debug code?

File: src/components/lib/tag.vue (hunk 11 above)
Comment (zhangtianli2006) [reply src/components/lib/tag.vue]: fixed

File: src/components/lib/confirm.vue (hunk 10 above)
Comment (zhangtianli2006) [reply src/components/lib/confirm.vue]: fixed

**Pull request merged**